pkg/api/addons.go

```diff
@@ -962,6 +1009,13 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 			o.KubernetesConfig.Addons[i].Enabled = to.BoolPtr(true)
 		}
 	}
+
+	// Enable keyvault flexvolume addon during upgrade for 1.16 or greater scenarios, unless explicitly disabled
```

Comment:
I don't think we need any special logic here. We should just be able to use whatever value is already in the api model during upgrade.